x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissions.java

```diff
@@ -199,12 +198,16 @@ public static Automaton buildPermittedFieldsAutomaton(final String[] grantedFiel
      */
     public FieldPermissions limitFieldPermissions(FieldPermissions limitedBy) {
         if (hasFieldLevelSecurity() && limitedBy != null && limitedBy.hasFieldLevelSecurity()) {
+            // TODO: the automaton computation is not cached
             Automaton _permittedFieldsAutomaton = Automatons.intersectAndMinimize(getIncludeAutomaton(), limitedBy.getIncludeAutomaton());
```

Comment:
This TODO is to note down that the FieldPermissionsCache (mainly for caching automaton computations) is only used for non-limiting roles, thus it does not apply here yet.

Reply:
nit: can make the TODO read as an instruction for something to do i.e. TODO: cache automaton computation

Reply:
Updated as suggested. Thanks